laceworksdk/api/crud_endpoint.py

```diff
@@ -0,0 +1,118 @@
+# -*- coding: utf-8 -*-
+
+from laceworksdk.api.base_endpoint import BaseEndpoint
+
+
+class CrudEndpoint(BaseEndpoint):
+
+    def __init__(self,
+                 session,
+                 object_type,
+                 endpoint_root="/api/v2"):
+        """
+        :param session: An instance of the HttpSession class.
+        :param object_type: The Lacework object type to use.
+        :param endpoint_root: The URL endpoint root to use.
+        """
+
+        super().__init__(session, object_type, endpoint_root)
+
+    def create(self, **request_params):
+        """
+        A method to create a new object.
+
+        :param request_params: Request parameters.
+
+        :return response json
+        """
+
+        json = self.build_dict_from_items(
+            request_params
+        )
+
+        response = self._session.post(self.build_url(), json=json)
+
+        return response.json()
+
+    def get(self, id=None, resource=None, **request_params):
+        """
+        A method to get objects.
+
+        :param guid: A string representing the object ID.
+        :param type: A string representing the object resource type.
+        :param request_params: A dictionary of parameters to add to the request.
+
+        :return response json
+        """
+
+        params = self.build_dict_from_items(
+            request_params
+        )
+
+        response = self._session.get(self.build_url(id=id, resource=resource), params=params)
+
+        return response.json()
+
+    def search(self, json=None, **kwargs):
+        """
+        A method to search objects.
+
+        :param json: A dictionary containing the desired search parameters.
+            (filters, returns)
+
+        :return response json
+        """
+
+        # TODO: Remove this on v1.0 release - provided for back compat
+        if kwargs["query_data"]:
```

Comment:
maybe better to do 

```suggestion
        query_data = kwargs.get("query_data")
        if query_data:
            json = query_data
```

just to prevent issues with kwargs not having that set at all, since kwargs is simply a dictionary, and you could have KeyError raised here

Reply:
done